Makefile

```diff
@@ -0,0 +1,254 @@
+
+tarfiles = $(figures) $(tables) $(styles) $(bibs) $(source)
+
+# prefix for difference files
+DIFPRE ?= diff_
+# git repo branch to compare changes against
+BASEBRANCH ?= master
+# git repo branch we are compiling
+THISBRANCH := $(shell git branch --no-color 2> /dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/ \1/' | tr -d ' ')
+# temp file for the version to compare against
+DRAFT = $(main).$(BASEBRANCH)
+# base for difference files to generate
+DIFF = $(DIFPRE)$(THISBRANCH)--$(BASEBRANCH)
+
+
+maketargets := all authlist clean copy diff help $(main) update tar templates tidy touch
+.PHONY: $(maketargets)
+
+styleopts := aastex61 apj apjl emulateapj lsstdescnote mnras prd prl tex 
+help:
+	@echo "Usage: make [style=lsstdescnote] [localpip=F] <target(s)>\n Possible targets: $(maketargets) <style>\n  all: equivalent to 'make $(main) copy'\n  authlist: use mkauthlist to generate latex author/affiliation listing\n  clean: remove latex temporary files AND compiled outputs\n  copy: copy $(main).pdf to $(outname).pdf\n  diff: if we're in a git repo, use latexdiff to compare the current $(main).tex vs the $(BASEBRANCH) branch, and attempt to compile to $(DIFF).pdf\n  help: what you're looking at\n  $(main): compile $(main).tex\n  update: update desc-tex, mkauthlist, and templates repo\n  tar: tar up latex source files to $(outname).tar.gz\n  templates: download latest templates to templates/\n  tidy: delete latex temporary files\n  touch: touch $(main).tex to force a recompile\n  <style>: equivalent to 'make style=<style> $(main) copy tar' (see style option)\n Options for style: $(styleopts)\n  (Anything else results in generic latex)\n Options for localpip: T or F. Set to T to install mkauthlist in this directory rather than systemwide"
+
+
+# Interpret `make` with no target as `make tex` (a latex Note).
+# At present, if the default_format is anything other than
+# [apj|apjl|mnras|prd|prl|emulateapj], a latex Note is made.
+# In future, we could think of using `make` to eg run the ipynb
+# notebook and make PDF from the output, but this has not been
+# implemented yet.
+$(main): export flag = \def\flag{${style}}
+envflag = \\def\\flag{${style}}
+all: $(main) copy
+
+copy:
+	cp ${main}.pdf ${outname}.pdf
+
+touch:
+	touch ${main}.tex
+
+# if we are in a git repo, add as a submodule; otherwise clone
+$(DESCTEX):
+	if [ -d .git ]; then git submodule add $(DESCTEXORIGIN); else git clone $(DESCTEXORIGIN); fi
+
+#http://journals.aas.org/authors/aastex/linux.html
+#change the compiler call to allow a "." file
+# {% raw %}
+$(main) : $(DESCTEX) authlist
+	STYLEFLAG=$(envflag) latexmk $(main)
+#	latexmk -g -pdf  \
+	-pdflatex='openout_any=a pdflatex %O -interaction=nonstopmode "${flag}\input{%S}"'  \
+	${main}
+# {% endraw %}
+
+diff: $(DIFF).pdf
+$(DIFF).pdf: $(DIFF).tex
+	STYLEFLAG=$(envflag) latexmk $<
+$(DIFF).tex: $(DRAFT).tex $(main).tex
+	latexdiff --exclude-textcmd="multicolumn" $^ > $@
+$(DRAFT).tex:
+	git show $(BASEBRANCH):$(main).tex > $@
```

Comment:
I remembered to mention that in the help, at least, but nothing wrong with this suggestion.